File: BinarySearch.java
public class BinarySearch{
    public static int binSearch(int[] arr,int key){
        int min=0;
        int max=arr.length-1;
        int mid;
        while (max>=min){
          mid=(max+min)/2;
          if (key==arr[mid]){
            return mid;
          }
          else if (key>arr[mid]){
            min=mid+1;
          }
          else {
            max=mid-1;
          }
        }
        return -1;
      }
      public static void main(String[] args) {
        int[] arr={1,3,5,6,7,9,5,6,7,8};
        System.out.println(binSearch(arr,5));
        System.out.println(binSearch(arr,4));
      }
}

File: BinarySearch.py
# returns index of given number, or -1 if not present in array
def binarySort(sortedList,x):
  min=0
  max=len(sortedList)-1
  mid=0
  while min<=max:
    mid=(max+min)//2
    if sortedList[mid]<x:
      min=mid+1
    elif sortedList[mid]>x:
      max=mid-1
    else:
      return mid
  return -1


#                [0 1 2 3 4 5 6 7] 
print(binarySort([1,2,3,4,5,6,7,8] , 5))